### PR TITLE
Lab 05

### PR DESCRIPTION
**Pull request by jbitton**
This lab covered quite a variety of topics, both theoretical and practical. We define the various types of layers that can be added in a convolutional neural network and what are residual connections. We also discuss Information Bottleneck and its relationship to CNNs. On the more practical side, both autograd in PyTorch and the impact of dimensionality of data are discussed as well.

-- Lab 5 (26 Feb 2019) by Divyansh Khanna and Lind Xiao. Edited by Joanna Bitton.

### CODE REVIEW

**Comment by Atcold** [comment]
PR summary, author list, and date are missing.
Plz, edit the first message, similarly to how it's done in #1.

**Comment by jbitton** [comment]
Updated! Sorry about that

**Comment by Atcold** [comment]
Very good, thanks.

**Comment by Atcold** [comment]
Can you replace the `\href` with `\cite`?

**Comment by jbitton** [comment]
Just noticed your comment about the `\href` replacement. Should I make another PR?

**Comment by Atcold** [comment]
Yes, plz, and add those references to the `BIB` file.
Thanks.

File: labs/05/practice.tex [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+The intuition is that in low dimensions random vectors point roughly in same directions.
+However in high dimensions, they point orthogonal to each other.
+As the dimensionality of the data is increased, more points becomes further apart and there are less matches.
+
+\begin{figure}
+    \centering
+    \includegraphics[width=\textwidth]{labs/05/images/dim3.png}
+    \caption{Projecting p for A with dimensions (10, 3)}
+    \label{fig:my_label1}
```
Comment (Atcold) [inline labs/05/practice.tex:75]: Use meaningful names.

File: labs/05/practice.tex [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    \vspace{5mm}
+    \includegraphics[width=\textwidth]{labs/05/images/dim5.png}
+    \caption{Projecting p for A with dimensions (10, 5)}
+    \label{fig:my_label2}
```
Comment (Atcold) [inline labs/05/practice.tex:79]: Use meaningful labels.

File: labs/05/theory.tex [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+There are a decent amount of variants proposed in the research community which are based on skip connections.
+DenseNets (\href{https://arxiv.org/abs/1608.06993}{paper}) has several parallel skip connections within each block.
+
+Figure 1 (\href{https://arxiv.org/pdf/1712.09913.pdf}{original paper}) depicts how the use of residual connections makes the loss landscape smooth, which allows the network to converge.
+
+\begin{figure}
+    \centering
+    \includegraphics[width=\textwidth]{labs/05/images/loss.png}
+    \caption{ The loss surfaces of ResNet-56 with/without skip connections.}
+    \label{fig:my_label0}
```
Comment (Atcold) [inline labs/05/theory.tex:84]: Use proper labels.
